Maps

jjgomezcadenas opened the pull request:
Types and functions associated with maps have been updated to include mapinfo, a new field which describes (in a panda Series) the ranges, binning and run number of the map.

jahernando commented:
The functions are almost self-documented, but please add an extra comment on the headers to indicate what are their purposes.

It seems to me that there is still one function missing. The function that computes the lifetime, by fitting an exponential, but using as points [(z, ene)] the result of applying the fit to a gaussian into z-ranges. Then you can either compute the lifetime or lifetime map using the profile of the result of the gaussian fits into z-slices. Do you agree?

gonzaponte commented:
> The functions are almost self-documented, but please add an extra comment on the headers to indicate what are their purposes.

Ok.


> It seems to me that there is still one function missing. The function that computes the lifetime, by fitting an exponential, but using as points [(z, ene)] the result of applying the fit to a gaussian into z-ranges.  Then you can either compute the lifetime or lifetime map using the profile of the result of the gaussian fits into z-slices. Do you agree?

I am not completely sure what you mean, but
- You can't use gaussian fits for the lifetime map, there is not that much statistics.
- I think that

```
do_z_slices
fit_to_expo
```

is more readable than

```
fit_to_expo_the_result_of_doing_z_slices
```

jahernando commented:
In fact, if your distribution is gaussian, there is little difference in doing the fit to a gaussian or to compute the mean and std of the distribution. If there is no stats for one, there is no stats for the other.

But I agree with you, that is more readable if one applies the function one after the other, then the reader can follow better what the is done. 

Please add the comments, and let the users to compute the lifetime with the gaussian fit.

jahernando commented:
Approved!

jjgomezcadenas committed:
Modify ASectorMap to add mapinfo.

Previous version of ASectorMap contained only the maps themselves. We
add now a pandas series containing the ranges, bins, and run number.

jjgomezcadenas committed:
Extend read/write of maps.

We read/write and additional Series describing ranges, binning and run number

jjgomezcadenas committed:
Make mapinfo Optional.

mapinfo is eventually filled as a Series, but defined initially as
None, so an Optional type is used.

jjgomezcadenas committed:
Update map functions to extension of ASectorMap type.

ASectorMap has been extended with maptype. Related map functions are
updated to reflect that change.

jjgomezcadenas committed:
Add function add_mapinfo()

Function add_mapinfo() takes a map (an object of type ASectorMap) and
filles the variable mapinfo (a Series) with metadata information,
specifying the ranges, bins and run_number for the map.

jjgomezcadenas committed:
Add a test for mapinfo.

jjgomezcadenas committed:
Add a new map with mapinfo.
